src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -94,7 +107,23 @@ namespace Microsoft.Quantum.Convert {
     /// let bi2 = BoolArrayAsBigInt([false, false, false, false, false, false, false, true]); // Not padded -> -128
     /// ```
     function BoolArrayAsBigInt(a : Bool[]) : BigInt {
-        body intrinsic;
+        mutable val = 0L;
+        mutable arr = a;
+
+        if Length(arr) % 8 != 0 {
+            // Padding is needed.
+            set arr += [false, size = Length(arr) % 8];
+        }
+
+        let len = Length(arr);
+        for i in 0..(len - 2) {
+            set val += arr[i] ? 2L ^ i | 0L;
+        }
+        if arr[len - 1] {
+            set val -= 2L ^ (len - 1);
+        }
+
+        return val;
     }
```

Comment:
I'm not strong in Q# but I have an impression that if the 0-length array is passed as an arg
* it is not documented what should and will happen,
* line 122 (L122), fragment `arr[len - 1]` will access outside of the array, causing an undefined or unexpected behavior.

Reply:
Hmm, you're right. I thought about updating the padding logic to pad a zero length array to length 8, but then figured that's a lot of extra work to get to zero value. So I've added an if-statement around the logic that should satisfy this.